src/controllers/article.js

```diff
@@ -21,5 +22,29 @@ export default {
     const articleData = await Articles.createComment({ articleId, comment });
 
     return { status: 201, data: articleData, message: 'comment added succesfully' };
+  },
+
+  /**
+      * controller for adding bookmarks
+      *
+      * @function
+      *
+      * @param {request} request - express request object
+      *
+      * @return {Object} - callback that execute the createBookmark controller
+      */
+  createBookmark: async (request) => {
+    const { articleId } = request.params;
+    const userId = request.body.user || {};
+
+    await Articles.getExistingArticle(articleId);
+
+    const existingBookmark = await Bookmarks.getExistingBookmark(articleId);
```

Comment:
This code doesn't validate if the bookmark belongs to the user requesting for the bookmark
also, you don't need to verify the user as the auth middleware does that

Reply:
Thanks for pointing that out. I have refactored the code to reflect these changes.